server/src/main/java/org/elasticsearch/search/vectors/KnnVectorQueryBuilder.java

```diff
@@ -393,7 +394,7 @@ protected Query doToQuery(SearchExecutionContext context) throws IOException {
             ? Math.round(Math.min(NUM_CANDS_MULTIPLICATIVE_FACTOR * requestSize, NUM_CANDS_LIMIT))
             : numCands;
         if (fieldType == null) {
-            throw new IllegalArgumentException("field [" + fieldName + "] does not exist in the mapping");
+            return new MatchNoDocsQuery();
```

Comment:
@mayya-sharipova @benwtrent This is the line I changed to get `KnnVectorQueryBuilder` to respect `allowUnmappedFields`

Reply:
I reverted this change due to the test failures it was causing. I'll file an issue for it and we can fix it later.